content/tutorial-build-run-rippled-ubuntu.md

```diff
@@ -0,0 +1,173 @@
+
+For an overview of `rippled`, see [Operating rippled Servers](tutorial-rippled-setup.html).
+
+Use these instructions to build a `rippled` binary file and run it as a stock `rippled` server on Ubuntu 15.04 or later. These instructions were tested on Ubuntu 16.04 LTS.
+
+For information about building `rippled` for other platforms, see [Builds](https://github.com/ripple/rippled/tree/develop/Builds) in the `rippled` GitHub repository.
+
+
+## System Requirements
+
+Here are the **minimum** system requirements for building and running `rippled`.
+
+**_To build rippled:_**
+
+* RAM: 8GB
+
+**_To run rippled:_**
+
+* RAM: 4GB
+* CPU: 64-bit x86_64, 2+ cores
+* Disk: 50GB SSD with 500 IOPS for the database partition
+* Network: A fast network connection is preferable
+
+If you are running `rippled` in an Amazon EC2 instance, Ripple recommends selecting a **Type** of **_m3.large_**. If you choose to run a validating `rippled` server, you may need more resources.
+
+
+## 1. Build `rippled`
+
+These instructions use Ubuntu's APT (Advanced Packaging Tool) to install the software prerequisites you need to build and run `rippled`.
+
+1. Update the list of packages that are available for `apt-get` to install or upgrade.
+```
+sudo apt-get update
+```
+2. Upgrade currently installed packages.
+```
+sudo apt-get -y upgrade
+```
+3. Install Git.
+```
+sudo apt-get -y install git
+```
+4. Install SCons.
+```
+sudo apt-get -y install scons
+```
+5. Install Ctags.
+```
+sudo apt-get -y install ctags
+```
+6. Install `pkg-config`.
+```
+sudo apt-get -y install pkg-config
+```
+7. Install Protocol Buffers.
+```
+sudo apt-get -y install protobuf-compiler
+sudo apt-get -y install libprotobuf-dev
+```
+8. Install Secure Socket Layer (SSL) toolkit development files.
+```
+sudo apt-get -y install libssl-dev
+```
+9. Install `python-software-properties`.
+```
+sudo apt-get install -y python-software-properties
+```
+10. Install Boost.
+```
+sudo add-apt-repository -y ppa:boost-latest/ppa
+sudo apt-get -y update
+sudo apt-get install -y libboost-all-dev
+```
+11. Get `rippled` source code.
+```
+git clone https://github.com/ripple/rippled.git
+cd rippled
+git checkout master
+```
+12. Build `rippled` binary executable from source code. This may take about 30 minutes.
+
+    SCons saves the built executable in `rippled/build`.
+
+```
+scons
+```
+
+
+## 2. Configure `rippled`
+
+1. Configure `rippled.cfg`.
+
+    `rippled.cfg` is the configuration file for `rippled`.
+
+    You can find the `rippled-example.cfg` example file in `rippled/doc`. See the file for a description of all configuration options. You should be able to connect to the XRP Ledger using the default configuration defined in the example file.
+
+    Put your configured `rippled.cfg` file in `etc/opt/ripple/`.
```

Comment:
what needs to change about the configuration to run a standalone server? I ask because if it is a simple configuration change - we may want to give the user the instructions on how to configure and run either flavor --

Reply:
Yes, sorry for the incorrect language. I am asking about stand-alone mode. 😄  How do you put your `rippled` server in stand-alone mode? Sorry if this is a dumb question - but does stand-alone mode mean that my `rippled` server still connects to the XRP Ledger, but not to any other `rippled` servers in the network? And this way, I can still query the XRP Ledger using the `rippled` API.

Reply:
The short answer is, you just add `-a` to the commandline to start your `rippled` server in stand-alone mode.

And I should probably schedule some time to explain some higher-level concepts of the XRP Ledger. The short version is, the servers in the network _are_ the XRP Ledger (sort of), so it doesn't make sense to ask if it connects to one but not the other. I mean, really, the XRP Ledger is just a concept, the shared state created out of the imagination of a bunch of servers dreaming in unison, and you can't truly connect to a concept.

Basically stand-alone mode is a lab setting where your server does all the imagining of the XRP Ledger all by its lonesome, so that you can do stuff like:

- submit transactions no one else will ever see
- freeze-frame, manually advance, and replay the ledger
- test out experimental code, force amendments to be on and see what they do
- explore alternate realities from a fresh starting point